src/main/java/com/mongodb/hibernate/internal/MongoAssertions.java

```diff
@@ -101,10 +101,11 @@ public static boolean assertFalse(boolean value) throws AssertionError {
         return false;
     }
 
-    public static <T> T assertInstanceOf(@Nullable Object value, Class<? extends T> type) {
-        if (!type.isInstance(value)) {
+    public static <T> T assertInstanceOf(Object value, Class<? extends T> type) {
+        try {
+            return type.cast(value);
+        } catch (ClassCastException e) {
             throw fail();
         }
-        return type.cast(value);
```

Comment:
`type.isInstance` was just duplicating the check that is anyway done by `type.cast`.